common/djangoapps/student/views/dashboard.py

```diff
@@ -863,6 +877,11 @@ def student_dashboard(request):  # lint-amnesty, pylint: disable=too-many-statem
         'resume_button_urls': resume_button_urls
     })
 
+    try:
+        context = DashboardRenderStarted.run_filter(context=context)
+    except DashboardRenderStarted.PreventDashboardRender as exc:
+        raise DashboardRenderNotAllowed(reverse(exc.redirect_to or 'account_settings')) from exc
```

Comment:
### What should happen when interacting with templates?

As it is right now, when filters that interact with templates (i.e they interact with template contexts) fail they raise an exception with the intent to redirect to some constant view (e.g if course about filter fails, then redirect to dashboard). If there's a handler that interprets this kind of exception, then the redirection occurs (as it happens when using [CourseAccessRedirect](https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/courseware/exceptions.py#L15) [here](https://github.com/openedx/edx-platform/pull/29840/files#diff-cfab8ddd186c3bde2aec4290ddf2926c4c00c1ee294571f67b969f2980ce6d97R1032) where [this handler](https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/courseware/views/views.py#L740) catches the exception and manages the redirection). If not, like in this case, then error 500 is thrown. So we can replicate this behavior we should implement a handler that manages exceptions in the student's dashboard, but that's probably out of the scope of this PR. 

Either way, should be this way the standard when interacting with templates?

Reply:
I like the idea that we should have a consistent way of interacting with templates when filters are involved. Having the ability to redirect somewhere and changing the template_name that should be rendered and naturally updating the context before rendering are the three that come to my mind now.